crate/cli/src/actions/cover_crypt/encrypt.rs

```diff
@@ -82,18 +102,151 @@ impl EncryptAction {
             .data
             .context("The encrypted data are empty")?;
 
-        // Write the encrypted file
-        let output_file = self
-            .output_file
-            .clone()
-            .unwrap_or_else(|| self.input_file.with_extension("enc"));
-        let mut buffer =
-            File::create(&output_file).with_context(|| "failed to write the encrypted file")?;
-        buffer
-            .write_all(&data)
-            .with_context(|| "failed to write the encrypted file")?;
-
-        println!("The encrypted file is available at {:?}", &output_file);
+        if let Some(CryptographicAlgorithm::CoverCryptBulk) = cryptographic_algorithm {
+            self.write_bulk_encrypted_data(&data)?;
+        } else {
+            // Write the encrypted file
+            let output_file = if let Some(output_file) = &self.output_file {
+                output_file.clone()
```

Comment:
You should be able to avoid cloning here since `File::create()` does not need it.

Reply:
yes but the fallback of this `if` cannot be a reference, hence I'm forced to clone here.

Reply:
True. You could have avoided it by duplicating the code outside your if condition... but anyway this is a path, so cloning shouldn't be expensive. I didn't see it, I thought this was the actual file bytes.

You could have used something like `self.output.cloned().unwrap_or(|| self.intputs.first()...)` to avoid the if. But this is only style.

Reply:
Ok in fact, I reworked this part in the same time you were reviewing, now it is the case:
https://github.com/Cosmian/kms/pull/37/files#diff-1254d8ddead7be701fb7158a92798d0b6b5f9ec46603fb9db99f5b502b4e4ba4R146